Decorator.cpp

```diff
@@ -0,0 +1,100 @@
+#include <iostream>
+#include <string>
+using namespace std;
+
+class Burger
+{
+public:
+    virtual string Serve() = 0;
```

Comment:
Please also add virtual destructor in this kind of cases
 `virtual ~Burger() = default;` and override it in derived classes  `~ClassicBurger() override = default;`